src/rest_vol_dataset.c

```diff
@@ -4363,3 +4393,111 @@ RV_convert_start_to_offset(hid_t space_id)
 
     return ret_value;
 } /* end RV_convert_start_to_offset() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_json_values_to_binary_callback
+ *
+ * Purpose:     A callback for RV_parse_response which will search
+ *              an HTTP response for the "value" field, and extract
+ *              the values into a newly allocated binary buffer.
+ *
+ *              Expects data in to be a pointer to an hid_t for the
+ *              datatype in the response, and data out to be the
+ *              address of a pointer that will point to the
+ *              newly allocated buffer.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ */
+static herr_t
+RV_json_values_to_binary_callback(char *HTTP_response, void *callback_data_in, void *callback_data_out)
```

Comment:
This function's fine for a first cut, but we'll definitely need much more sophisticated parsing eventually because it's likely that we're going to get many other things than just `H5T_NATIVE_INT/FLOAT/DOUBLE` for the mem type